test/test_nn.py

```diff
@@ -8682,6 +8706,28 @@ def test_cudnn_noncontiguous_weight(self):
         self.assertEqual(F.conv1d(input, weights1, bias=None, stride=2, dilation=2),
                          F.conv1d(input, weights2, bias=None, stride=2, dilation=2))
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    @unittest.skipIf(not TEST_CUDNN, "needs cudnn")
+    def test_conv_cudnn_nhwc(self):
+        input = torch.randint(1, 10, (2, 8, 4, 4), dtype=torch.float32, device="cuda").as_strided((2, 8, 4, 4), (128, 1, 32, 8), 0).requires_grad_(True)
```

Comment:
nit: `nhwc = torch.randn((10, 3, 32, 32), device=device).contiguous(memory_format=torch.channels_last)` gives you same result and more readable

Reply:
Will update.
I switched this to manual `as_strided` when I was having issue with the special cases back then.